docs/content/docs/build/container-states-events.md

```diff
@@ -0,0 +1,336 @@
+
+- A [container.connectionState]({{< relref "/docs/apis/fluid-static/ifluidcontainer-interface.md#connectionstate-propertysignature" >}}) property of type `ConnectionState`. There are four possible values for the property:
+
+    - `Disconnected` 
+    - `EstablishingConnection`: Your code should treat this state the same as it treats the disconnected state. See [Examples](#examples).
+    - `CatchingUp`: In most scenarios, your code should treat this state the same as it treats the connected state. See [Examples](#examples). An exception would be when it is important that users see the very latest changes from other clients before they are allowed to make their own edits. In that case, your code should treat **catching up** like it treats the **disconnected** state.
+    - `Connected` 
+    
+- A *disconnected* event, that fires if a network problem causes a disconnection or if the `container.disconnect` method is called.
+- A *connected* event, that fires if the Fluid client runtime is able to reconnect (and any needed catching up is complete) or if the `container.connect` method is called.
+
+##### Examples
+
+Your code can disable the editing UI in your application when the container is disconnected, and then reenable it when a connection is restored, as shown in the following example:
+
+```typescript
+container.on("disconnected", () => {
+    // Prevent user edits to disable data loss.
+});
+
+container.on("connected", () => {
+    // Enable editing if disabled.
+});
+```
+
+Your code can reduce unnecessary network traffic by disconnecting when a user is idle. The following example assumes that there is a `user` object that emits *idle* and *active* events. 
+
+```typescript
+user.on("idle", () => {
+    // Disconnect the container when the user is idle.
+    container.disconnect();
+});
+
+user.on("active", () => {
+    // Connect the container when the user is active again.
+    container.connect();
+});
+```
+
+### Local Readiness state
+
+Local Readiness is a *client-relative state*: the container may have a different Local Readiness state on different clients. The following diagram shows the possible Local Readiness states and the events that cause a state transition. Details are below the diagram.
+
+<!-- TO MODIFY THIS DIAGRAM, SEE INSTRUCTIONS AT THE BOTTOM OF THIS FILE. -->
+![A state diagram of the two possible Local Readiness states](./images/LocalReadinessStates.png)
+
+#### Ready
+
+In this state, changes to the shared data objects can be made locally. They are shared with the Fluid service when, and only when, the container is in **published** and **connected** status. 
+
+A container is automatically in **ready** state as soon as it is created and it remains in this state unless it is disposed. 
+A disposed container transitions back to **ready** if its ID is passed to a call of `client.getContainer`.
```

Comment:
The specific instance of the FluidContainer remains disposed forever - probably clarify that this isn't a transition, but rather just that a new "ready" container can be retrieved if needed.
```suggestion
A disposed `FluidContainer` remains disposed forever, but a new **ready** `FluidContainer` can be created by passing its ID to a call of `client.getContainer`.
```